packages/project-access/src/file/file-search.ts

```diff
@@ -68,21 +113,33 @@ export function findBy(options: {
             }
         });
         finder.on('file', (file: string) => {
-            if (extensionNames.includes(extname(file)) || fileNames.includes(basename(file))) {
+            if (
+                extensionNames.includes(extname(file)) ||
+                fileNames.includes(basename(file)) ||
+                (fileNames.length === 0 && extensionNames.length === 0)
+            ) {
                 results.push(file);
             }
         });
         finder.on('end', () => {
-            let searchResult: string[] = results;
-            if (options.memFs) {
-                const { modified, deleted } = getMemFsChanges(options.memFs.dump(''), fileNames, extensionNames);
-                const merged = Array.from(new Set([...results, ...modified]));
-                searchResult = merged.filter((f) => !deleted.includes(f));
+            const { searchResult, fatalErrors } = getFindResultOnEnd(
+                results,
+                fileNames,
+                extensionNames,
+                options.root,
+                errors,
+                options.memFs
+            );
+
+            if (fatalErrors.length === 0) {
+                resolve(searchResult);
+            } else {
+                // eslint-disable-next-line prefer-promise-reject-errors
+                reject(fatalErrors);
```

Comment:
@heimwege, need you opinion on this, because I was a bit struggling. You think it is of to return an array of errors? Or should be check if it just one and handle it somehow different in case multiple fatal errors exist.